### ISSUE TITLE
[Feature Request] Support S3 Storage

### ISSUE THREAD

**Issue by MFYDev** [karakeep-app/karakeep#305]
Hi there,

Thank you for the great project, it is truly amazing of using Hoarder. 

However, as time goes by, I found that I need larger storage, then I started to selfhosted Minio on my HDD, which can give me more disk storage. I found that Hoarder does not support S3 yet.

It will be so amazing that if Hoarder can support S3 storage and especially include S3 path style with it. So that I do not need to worry about the storage space anymore

**Comment by MohamedBassem** [comment]
Yeah, this probably makes sense at some point. Though to be honest with you, this might take some time to be picked up compared to all the other stuff that we need to work on.

**Comment by jpmurray** [comment]
I'm curently testing what will be my prefered service (so far between Hoarder and Linkwarden), and this would be the only thing needed to make me happy.

Being able to offload storage of saved media to an S3 compatible endpoint would enable me to run Hoarder in a small homelab without the need to have "a lot" of storage for the peace of mind...

**Comment by maelp** [comment]
Would be really nice to have S3 storage for videos, etc

**Comment by maelp** [comment]
BTW possibly using something like s3fs would allow putting stuff in s3 without having to deal with a complex implementation? eg the user just uses s3fs to create a link to the bucket, and Karakeep puts stuff there?

https://medium.com/@cbates255/docker-container-with-bind-mount-to-s3-1d000486d44d
https://github.com/s3fs-fuse/s3fs-fuse
